### ISSUE TITLE
Multi-line Code blocks from WP 5.5 lose line breaks when GB active

### ISSUE THREAD

**Issue by westonruter** [WordPress/gutenberg#26301]
Given an Code block with the content:

```html
<!-- wp:code -->
<pre class="wp-block-code"><code>if ( a === b ) {
    doSomething();
}</code></pre>
<!-- /wp:code -->
```

In WP 5.5 when Gutenberg is inactive, this is rendered in the editor as follows:

![image](https://user-images.githubusercontent.com/134745/96514944-4c637180-1219-11eb-8a95-e55ca1d629c0.png)

When activating Gutenberg, however, this gets rendered in the editor as:

![image](https://user-images.githubusercontent.com/134745/96515003-5f764180-1219-11eb-8387-be25654b2df5.png)

This appears to be a regression introduced in v9.1 with allowing HTML editing and rich text in the Code block: https://github.com/WordPress/gutenberg/pull/24689, since the block's `content` attribute changed from a `text` source to an `html` source. This is in spite of the fact that the code is still in a `pre`.

If I try recreating this Code block with Gutenberg active, then I get this saved content:

```html
<!-- wp:code -->
<pre class="wp-block-code"><code>if ( a === b ) {<br>    doSomething();<br>}</code></pre>
<!-- /wp:code -->
```

Note that `<br>` tags are now needed to represent line breaks instead of newline characters.

Is there a block deprecation function missing here?

cc @ellatrix @mkaz

**Comment by getdave** [comment]
I can confirm this is happening on my personal site as well.

**Comment by joyously** [comment]
I was testing something else, but saw this reformatting happening in the Classic block as well, which I don't think was happening before.

**Comment by talldan** [comment]
#26347 seems to fix a similar issue.

**Comment by tellthemachines** [comment]
Closing as fixed by #26347.
